services/core-web/common/actionCreators/orgbookActionCreator.js

```diff
@@ -0,0 +1,34 @@
+import { showLoading, hideLoading } from "react-redux-loading-bar";
+import { request, success, error } from "../actions/genericActions";
+import * as reducerTypes from "../constants/reducerTypes";
+import * as orgbookActions from "../actions/orgbookActions";
+import * as API from "../constants/API";
+import { ENVIRONMENT } from "../constants/environment";
+import { createRequestHeader } from "../utils/RequestHeaders";
+import CustomAxios from "../customAxios";
+
+export const searchOrgBook = (search) => (dispatch) => {
+  dispatch(request(reducerTypes.ORGBOOK_SEARCH));
+  dispatch(showLoading());
+  return CustomAxios()
+    .get(ENVIRONMENT.apiUrl + API.ORGBOOK_SEARCH(search), createRequestHeader())
+    .then((response) => {
+      dispatch(success(reducerTypes.ORGBOOK_SEARCH));
+      dispatch(orgbookActions.storeSearchOrgBookResults(response.data));
+    })
+    .catch(() => dispatch(error(reducerTypes.ORGBOOK_SEARCH)))
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const fetchOrgBookCredential = (credentialId) => (dispatch) => {
```

Comment:
I thought we were going through our API as a proxy for calls to orgbook. Why does the front-end need a credential?

Reply:
The search field uses a search endpoint that doesn't return anything useful except the name of the entity and its latest credential ID. Using the latest credential ID, we can make an API call (fetchOrgBookCredential) to get it's source_id, which is needed to create the URL for the **View on OrgBook** button, as well as to display any other additional info we may want to show the user in the UI before they confirm and click the **Associate** button.

Reply:
Credential is VON name for the record, not a credential for the actual web service.